### DISCUSSION TITLE
Moving to Kubernetes SIGs, CNCF CLA

### DISCUSSION THREAD

**Discussion by shaneutt** [kubernetes-sigs/blixt#42] [Announcements]
Hi everyone :wave: 

Several weeks ago at [Kong](https://konghq.com) we decided we wanted to donate this load-balancer to [Kubernetes SIGs](https://github.com/kubernetes-sigs) for use in upstream [Gateway API](https://github.com/kubernetes-sigs/gateway-api).

We've been approved for contribute this repository to the [CNCF](https://cncf.io) and [Kubernetes SIGs](https://github.com/kubernetes-sigs):

https://github.com/kubernetes/org/issues/3875

What follows next are some formalities, some of which we'll need contributor participation for.

### Licensing Review

We're obligated to use a [GPL](https://en.wikipedia.org/wiki/GNU_General_Public_License) compatible license for parts of our code because we use [eBPF](https://ebpf.io/) for the data-plane and link to Linux kernel code. This however needs to undergo legal review in the CNCF because this is generally not a well-trodden path:

https://github.com/cncf/foundation/issues/474

### CNCF Contributor License Agreement

Everyone who has contributed to the repository so far will need to have signed the [CNCF Contributor License Agreement (CLA)](https://github.com/cncf/cla) before we can migrate. This can be done by creating any PR in any Kubernetes SIGs repository, and the bot will send you a link if you haven't already signed it.

The following is a list tracking contributors and whether or not they've signed the CNCF CLA:

- [x] @acald-creator 
- [x] @alejandroEsc
- [x] @astoycos 
- [x] @mlavacca 
- [x] @Rahul-D78 
- [x] @rushi47 
- [x] @shaneutt 
- [x] @synthe102 
- [x] @wheelerlaw 
- [x] @levikobi
~@RudranshGoel~

Please feel free to reach out to me directly if you have any trouble with this or any questions, I can be reached on Kubernetes Slack `@shane`, OR feel free to add a comment here in this discussion letting us know if/when you've signed the CLA.

If you haven't signed the CLA and we don't get a response from you in the next few weeks, OR if for some reason you're opposed to signing the CLA just note that we may have to drop your commits from the `main` branch. Not something we want to do, but this is the fallback position for compliance.

Thanks all, will continue to post updates about the progress of things, but also feel free to subscribe to these:

- https://github.com/kubernetes/org/issues/3875
- https://github.com/cncf/foundation/issues/474

And let us know in this discussion if you have any questions.

**Comment by shaneutt** [comment]
Here's an example of how @wheelerlaw went about triggering a CLA sign for himself:

https://github.com/kubernetes-sigs/gateway-api/pull/1588#issuecomment-1343332688

Feel free to do the same: create a draft PR on https://github.com/kubernetes-sigs/gateway-api either with a real change you want to contribute, or something we can just close once we're done.

OR if you already signed but I haven't counted you in the list yet, just link a previous CNCF PR of yours.

**Reply by shaneutt** [reply]
@acald-creator is now all set: https://github.com/kubernetes-sigs/gateway-api/pull/1592

**Reply by shaneutt** [reply]
@alejandroEsc is all set https://github.com/kubernetes-sigs/cluster-api/pull/6188

**Comment by Rahul-D78** [comment]
Hi @shaneutt am I all set -- https://github.com/kubernetes-sigs/gateway-api/pull/1593

**Comment by shaneutt** [comment]
@RudranshGoel just wanted to check in on this, please let us know if you can or have already signed the CNCF CLA (by sharing a PR).

If we don't hear back from you in the coming weeks, the most likely action will be we'll have to drop or revert your commits from the repository before we transfer. Please check in if you can!

**Reply by shaneutt** [reply]
@RudranshGoel just wanted to give you a heads up that we expect things to be moving soon, so please note that for compliance reasons we will be dropping commits authored by you if we don't hear back from you soon. Let us know if there's any confusion, and if we can help!

**Reply by shaneutt** [reply]
@RudranshGoel just wanted to provide one more warning that if we don't hear back from you on this one, we may need to drop your contributions before we can migrate to the CNCF. If you get this message, please read this announcement over and let us know if you have any questions, concerns or problems we can assist with.

**Comment by shaneutt** [comment]
@synthe102 if I'm not mistaken you are already signed up for CNCF contributions? Just share a relevant PR here if so, and if not feel free to follow the "test PR" approach some used above.

**Reply by synthe102** [reply]
Yes I signed up, the CLA check is valid on this PR: https://github.com/Kong/blixt/pull/61

**Reply by synthe102** [reply]
That's better: https://github.com/kubernetes-sigs/gateway-api/pull/1720

**Reply by shaneutt** [reply]
Awesome, thanks!

**Comment by shaneutt** [comment]
@levikobi if you could link to an upstream Kubernetes or CNCF PR prior to your contribution in https://github.com/Kong/blixt/pull/93, (OR just use the "test PR" approach as seen above) it would be greatly appreciated as we should be really close to the point where https://github.com/cncf/foundation/issues/474 should be done and we should be starting the CNCF migration to ksigs.

**Reply by levikobi** [reply]
Sure, https://github.com/kubernetes-sigs/gateway-api/pull/2179

**Reply by shaneutt** [reply]
Thank you!

**Comment by shaneutt** [comment]
https://github.com/Kong/blixt/discussions/42 has the potential to help move this along, if the vote there to graduate Cilium (which shares a similar licensing schema to us) succeeds, just a heads up for everyone that we're still waiting, but it's possible after this is resolved we could get moving finally.

**Comment by shaneutt** [comment]
Licensing as been approved: https://github.com/cncf/foundation/issues/474 the process should be able to continue now!

**Comment by shaneutt** [comment]
The migration is now fully complete, thanks everyone!
